File: RunnableHandler/app/src/main/java/com/example/runnablehandler/MainActivity.java
package com.example.runnablehandler;

import android.annotation.SuppressLint;
import android.os.Bundle;
import android.os.Handler;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;

import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;

public class MainActivity extends AppCompatActivity {

    TextView timerTextView;
    Button startButton;
    Button stopButton;
    Runnable runnable;
    Handler handler = new Handler();
    int seconds = 0;

    @SuppressLint("MissingInflatedId")
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        EdgeToEdge.enable(this);
        setContentView(R.layout.activity_main);
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main), (v, insets) -> {
            Insets systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars());
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom);
            return insets;
        });

        timerTextView = findViewById(R.id.timerTextView);
        startButton = findViewById(R.id.startButton);
        stopButton = findViewById(R.id.stopButton);
    }

    public void start(View view){
        handler = new Handler();
        runnable = new Runnable() {
            @Override
            public void run() {
                timerTextView.setText("Timer: " + seconds);
                seconds++;
                timerTextView.setText("Timer: " + seconds);
                handler.postDelayed(runnable, 1000);
            }
        };
        handler.post(runnable);
        startButton.setEnabled(false);
        stopButton.setEnabled(true);
    }

    public void stop(View view){
        handler.removeCallbacks(runnable);
        startButton.setEnabled(true);
        stopButton.setEnabled(false);
        seconds = 0;
        timerTextView.setText("Timer: " + seconds);
    }
}

File: My-Own-Projects/LandmarkBook/app/src/main/java/com/farukcesur/landmarkbook/MainActivity.kt
package com.farukcesur.landmarkbook

import android.content.Intent
import android.os.Bundle
import android.widget.AdapterView
import android.widget.ArrayAdapter
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat
import com.farukcesur.landmarkbook.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {

    private lateinit var binding: ActivityMainBinding
    private lateinit var landmarkList: ArrayList<Landmark>

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(layoutInflater)
        val view = binding.root
        setContentView(view)
        enableEdgeToEdge()
        ViewCompat.setOnApplyWindowInsetsListener(view) { v, insets ->
            val systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars())
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom)
            insets
        }

        landmarkList = ArrayList<Landmark>()

        val istanbul = Landmark("İstanbul", "Türkiye", R.drawable.istanbul)
        val konya = Landmark("Konya", "Türkiye", R.drawable.konya)
        val kabe = Landmark("Kabe", "Arabistan", R.drawable.kabe)
        val medine = Landmark("Medine", "Arabistan", R.drawable.medine)

        landmarkList.add(istanbul)
        landmarkList.add(konya)
        landmarkList.add(kabe)
        landmarkList.add(medine)

        val adapter = ArrayAdapter(this, android.R.layout.simple_list_item_1, landmarkList.map { landmark -> landmark.name })
        binding.listView.adapter = adapter

        binding.listView.onItemClickListener =
            AdapterView.OnItemClickListener { parent, view, position, id ->
                val intent = Intent(MainActivity@ this, DetailsActivity::class.java)
                intent.putExtra("landmark", landmarkList.get(position))
                startActivity(intent)
            }
    }
}